File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        n = len(nums)
        nums.sort()
        mini = nums[0]
        left = 0
        maxi = float('-inf')
        maxDiff = float('inf')
        for right in range(n):
            maxi = max(maxi, nums[right])
            while right - left + 1 > k:
                left+=1 
            if right - left + 1 == k:
                mini = nums[left]
                maxDiff = min(maxDiff, maxi - mini)
        return maxDiff

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int minDiff = Integer.MAX_VALUE;
        for(int i=k-1; i<nums.length; i++){
            minDiff = Math.min(minDiff, nums[i] - nums[i-k+1]);
        }
        return minDiff == Integer.MAX_VALUE ? 0 : minDiff;
    }
}